src/infrastructure/services/blockstack-service.ts

```diff
@@ -94,4 +110,132 @@ export class BlockstackService {
             return registeredDomainArray[0];
         }
     }
+    public getAddressMap = async (blockstackID: BlockstackId, tag?: string): Promise<IAddressMapping> => {
+        const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(blockstackID);
+        return getContentFromGaiaHub(blockstackID.toString(), cruxPayFileName, this.bnsNodes, tag, this.cacheStorage);
+    }
+    public putAddressMap = async (addressMapping: IAddressMapping, cruxDomainId: CruxDomainId, keyManager: IKeyManager): Promise<string> => {
+        const blockstackID = await this.getBlockstackIdFromKeyManager(keyManager, cruxDomainId);
+        if (!blockstackID) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.UserDoesNotExist);
+        }
+        const addressDecoder: Decoder<IAddress> = object({
+            addressHash: string(),
+            secIdentifier: optional(string()),
+        });
+        try {
+            for ( const currency of Object.keys(addressMapping) ) {
+                const addressObject: IAddress = addressDecoder.runWithException(addressMapping[currency]);
+            }
+        } catch (error) {
+            throw ErrorHelper.getPackageError(error, PackageErrorCode.AddressMappingDecodingFailure);
+        }
+        const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(blockstackID);
+        const gaiaDetails = await getGaiaDataFromBlockstackID(blockstackID.toString(), this.bnsNodes, undefined, this.cacheStorage);
+        const finalURL = await new GaiaService(gaiaDetails.gaiaWriteUrl).uploadContentToGaiaHub(cruxPayFileName, addressMapping, keyManager);
+        log.info(`Address Map for ${blockstackID} saved to: ${finalURL}`);
+        return finalURL;
+    }
+    public getBlockstackIdFromKeyManager = async (keyManager: IKeyManager, cruxDomainId: CruxDomainId): Promise<BlockstackId|undefined> => {
+        const userSubdomainOwnerAddress = publicKeyToAddress(await keyManager.getPubKey());
+        const registeredBlockstackIDs = await this.getRegisteredIDsByAddress(userSubdomainOwnerAddress);
+        console.log(IdTranslator.cruxToBlockstack(cruxDomainId).toString());
```

Comment:
modify `console.log` to either `log.info()` or `log.debug()`

Reply:
everywhere in the SDK

Reply:
Removed was there for local debugging